src/Ledgers.jl

```diff
@@ -12,94 +12,119 @@ Licensed under MIT License, see LICENSE.md
 module Ledgers
 
 using UUIDs, StructArrays, AbstractTrees
-using Instruments, Assets
+using Instruments
 import Instruments: instrument, symbol, amount, name, currency
+using Assets: USD
 
-export Credit, Debit, Account, Ledger, Entry, AccountId, AccountInfo
-export id, balance, credit!, debit!, post!, instrument, symbol, amount, name, currency
+export Account, Ledger, Entry, AccountId, AccountCode, AccountInfo, AccountGroup
+export id, balance, credit!, debit!, post!, instrument, symbol, amount, code, name, currency
+export parent, subaccounts, subgroups
 
 abstract type Identifier end
+
 struct AccountId <: Identifier
     value::UUID
 end
+
 AccountId() = AccountId(uuid4())
 
 struct AccountCode
     value::String
 end
 
-abstract type AccountType end
-struct Credit <: AccountType end
-struct Debit <: AccountType end
+abstract type AccountType{B <: Position} end
 
-mutable struct Account{P<:Position}
+mutable struct Account{B <: Position} <: AccountType{B}
```

Comment:
I'm not so sure this should be `mutable`.
That would mean that instead of a vector of the actual Int64 numbers, you'd have a vector of pointers to this structure, I think.   We need to look carefully at what StructArrays does do this.

Reply:
I'm up for making `Account` immutable, but would that impact all the references?

For example, `AccountInfo` points to a mutable account. If `Account` is immutable, do we need to update `AccountInfo` every time the balance changes?

Reply:
This is where I'd said before, it wasn't like what we'd talked about.
AccountInfo should really reference the Ledger, and the index within that ledger, to access the actual balance of an account.